### PR TITLE
Codegen Migration of Admin REST APIs

### PR DESCRIPTION
**Pull request by hisanhunais**
## Purpose
A code generator is used to generate server-side code skeletons using a swagger specification. Currently, a proprietary code generator plugin is used for Admin REST APIs. This PR migrates the code generator of the Admin REST APIs to the [swagger-codegen](https://github.com/swagger-api/swagger-codegen). 

Resolves: https://github.com/wso2/product-apim/issues/7977

## Goals
1. Migrate the code generator of the Admin REST APIs to the swagger-codegen.
2. Migrate the scopes from x-scopes to default swagger specification supported scopes in the admin-api.yaml file.
3. Transfer the Implementation and Util codes of previous Admin REST APIs appropriately.

## Approach
- Created a new directory as org.wso2.carbon.apimgt.rest.api.admin.**v1** under components/apimgt. (Duplicated the org.wso2.carbon.apimgt.rest.api.admin directory and removed unnecessary files and directories).
- Added the swagger-codegen-maven-plugin to the pom.xml of org.wso2.carbon.apimgt.rest.api.admin.**v1**.  Slightly modified [code templates](https://github.com/wso2/carbon-apimgt/tree/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.util/src/main/resources/server-templates) were used to generate the relevant code segments instead of the default code templates used by swagger-codegen. 
- Migrated the scopes from x-scopes to default swagger specification supported scopes in admin-api.yaml 
- Built org.wso2.carbon.apimgt.rest.api.admin.**v1**. This creates the necessary server-side code skeletons and DTOs under src/gen/java.
- Copied the utils codes from org.wso2.carbon.apimgt.rest.api.admin directory to org.wso2.carbon.apimgt.rest.api.admin.**v1** diretory and made relevant changes.
- Copied the Implementation codes from org.wso2.carbon.apimgt.rest.api.admin directory to org.wso2.carbon.apimgt.rest.api.admin.**v1** directory and made relevant changes.
- Enabled obtaining admin resource mappings for admin v1 REST API which would be used for scope validation.

## Test environment
JDK 1.8.0_241
Ubuntu 18.04.4 LTS

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ /*
+  ~  * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/WEB-INF/web.xml:4]: Please fix the license header.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/WEB-INF/beans.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/WEB-INF/beans.xml:1]: Add license header

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/META-INF/webapp-classloading.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
+
+<!--
+  ~ Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/webapp/META-INF/webapp-classloading.xml:4]: Please fix the license header.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/v1/ApplicationsApi.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+@Produces({ "application/json" })
+
+
+public class ApplicationsApi  {
+
+  @Context MessageContext securityContext;
+
+ApplicationsApiService delegate = new ApplicationsApiServiceImpl();
+
+
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/v1/ApplicationsApi.java:38]: shall we remove this extra line?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/.swagger-codegen-ignore [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# You can also negate patterns with an exclamation (!).
+# For example, you can ignore all files in a docs folder with the file extension .md:
+#docs/*.md
+# Then explicitly reverse the ignore rule for a single file:
+#!docs/README.md
+
+**/RestApplication.java
+**/test/**
+swagger.json
+pom.xml
```
Comment (ruwiniwj) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/.swagger-codegen-ignore:28]: is this an autogenerated file? if not don't we need a new line eof?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/v1/ApplicationsApi.java (hunk 4 above)
Comment (ruwiniwj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/gen/java/org/wso2/carbon/apimgt/rest/api/admin/v1/ApplicationsApi.java:38]: i think this is an autogenerated file. In that case we don't modify the file after generation.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApiCategoriesApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                                                  MessageContext messageContext) {
+        try {
+            APIAdmin apiAdmin = new APIAdminImpl();
+            String userName = RestApiUtil.getLoggedInUsername();
+            String tenantDomain = MultitenantUtils.getTenantDomain(userName);
+            int tenantID = APIUtil.getTenantIdFromTenantDomain(tenantDomain);
+            APICategory apiCategoryToUpdate = APICategoryMappingUtil.fromCategoryDTOToCategory(body);
+            APICategory apiCategoryOriginal = apiAdmin.getAPICategoryByID(apiCategoryId);
+            if (apiCategoryOriginal == null) {
+                String errorMsg = "No api category with the given category ID exists :" + apiCategoryId;
```
Comment (Meruja) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApiCategoriesApiServiceImpl.java:88]: Shall we change 'api' as 'API'?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/LabelsApiServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+package org.wso2.carbon.apimgt.rest.api.admin.v1.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.cxf.jaxrs.ext.MessageContext;
+import org.wso2.carbon.apimgt.api.APIAdmin;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.Label;
+import org.wso2.carbon.apimgt.impl.APIAdminImpl;
+import org.wso2.carbon.apimgt.rest.api.admin.v1.*;
```
Comment (Chamindu36) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/LabelsApiServiceImpl.java:26]: Shall we remove the wildcard imports?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/FileBasedApplicationImportExportManager.java [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import java.util.zip.ZipOutputStream;
```
Comment (Chamindu36) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/FileBasedApplicationImportExportManager.java:41]: Shall we remove the wildcard imports here?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/.swagger-codegen-ignore (hunk 5 above)
Comment (hisanhunais) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/.swagger-codegen-ignore:28]: This file has been removed in the current master branch.
https://github.com/wso2/carbon-apimgt/tree/master/components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApiCategoriesApiServiceImpl.java (hunk 6 above)
Comment (hisanhunais) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/ApiCategoriesApiServiceImpl.java:88]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/f704b669e66b63967e82ac525240beebdeed9b05

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/LabelsApiServiceImpl.java (hunk 7 above)
Comment (hisanhunais) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/LabelsApiServiceImpl.java:26]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/5f809500abaaa925adebc219c361dba658570994

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/FileBasedApplicationImportExportManager.java (hunk 8 above)
Comment (hisanhunais) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/FileBasedApplicationImportExportManager.java:41]: Fixed with https://github.com/wso2/carbon-apimgt/pull/9702/commits/5f809500abaaa925adebc219c361dba658570994

**Pull request merged**